packages/suite/src/middlewares/firmware/firmwareMiddleware.ts

```diff
@@ -1,21 +1,27 @@
 import { SUITE } from 'src/actions/suite/constants';
-import { FIRMWARE } from 'src/actions/firmware/constants';
-import * as firmwareActions from 'src/actions/firmware/firmwareActions';
+import {
+    firmwareActions,
+    firmwareUpdate,
+    validateFirmwareHash,
+} from 'src/actions/firmware/firmwareActions';
 import type { MiddlewareAPI } from 'redux';
 import type { AppState, Action, Dispatch } from 'src/types/suite';
 
-const firmware =
+import { selectFirmware } from '../../reducers/firmware/firmwareReducer';
+
+const firmwareMiddleware =
     (api: MiddlewareAPI<Dispatch, AppState>) =>
     (next: Dispatch) =>
     (action: Action): Action => {
-        const { firmware, router } = api.getState();
+        const firmware = selectFirmware(api.getState());
```

Comment:
is that somehow better than destructuring it? 🤔

Reply:
We introduced linter rule in `eslintrc.js` so linter complains about using `getState` directly so I used the selector instead.
<img width="1313" alt="image" src="https://github.com/trezor/trezor-suite/assets/36101761/11b412eb-b0d8-41f0-90cd-844c6c7853a9">

Reply:
but `getState()` doesn't seem to be typed as `any`, I'm slightly confused here

Reply:
Yes, in suite desktop the `getState` function is typed properly but that is not the case in `suite-common` where we will use `createThunk` function that will mean `getState` return is `any`. So I just preferred to use selectors.

Also if we start refactoring data structure or something, we will only need to change that in one place where the selector has been created. 

I believe that our goal was to use selectors as much as possible instead of destructuring state. 

Maybe @Nodonisko can elaborate?

Reply:
This is also relevant for:
https://github.com/trezor/trezor-suite/pull/8864/files/57913fc7c81829cafb80b5f76a16b81991967886#diff-545d00698b46c7b1a497151825b00ef48aea064f955cf81b2526e8451dfc9458

Reply:
fair! Still a bit strange to use 4 selectors instead of selecting just 1 object? But leaving that up to you guys :)

Reply:
Never access any property from state without selector even simple properties like `state.firmware`. Google for redux lookup selectors for more info.

Reply:
googling redux lookup selectors didn't find me anything relevant, could you please help me find some useful link?

Reply:
Something is here: https://redux.js.org/usage/deriving-data-selectors#encapsulating-state-shape-with-selectors
Long story short there are two main reason:
1. easier refactoring of state shape
2. type-safety - in suite-common `getState` returns `any` so using it only in selectors ensures type-safety